test_aodndata/acorn/test_acorn.py

```diff
@@ -0,0 +1,84 @@
+import os
+from datetime import datetime
+
+from aodncore.pipeline import PipelineFilePublishType
+from aodncore.pipeline.exceptions import InvalidFileNameError
+from aodndata.acorn.handler import AcornHandler
+from aodncore.testlib import HandlerTestCase
+
+TEST_ROOT = os.path.join(os.path.dirname(__file__))
+
+GOOD_NC_FV00 = os.path.join(TEST_ROOT,
+                       'IMOS_ACORN_V_20180501T003000Z_ROT_FV00_1-hour-avg.nc')
+
+GOOD_NC_FV01 = os.path.join(TEST_ROOT,
+                       'IMOS_ACORN_V_20170101T003000Z_ROT_FV01_1-hour-avg.nc')
+
+NON_INDEX_NC = os.path.join(TEST_ROOT,
+                            'IMOS_ACORN_V_20101201T050000Z_BONC_FV00_sea-state.nc')
+
+
+BAD_FILES = ('IMOS_ACON_RV_20171014T060000Z_LANC_FV00_radial.nc',
+             'IMOS_ACORN_RV_201611203000Z_GUI_FV00_radial.nc',
+             'IMOS_ACORN_V_20180910T010000Z_BONC_FV00_sea-se.nc',
+             'IMOS_ACORN_RV_20120507T053500Z_FV01_radial.nc',
+             'IMOS_ACORN_RV_20191014T192000Z_TAN_FV01_radial.nc.some_suffix',
+             'prefix.IMOS_ACORN_RV_20191014T192000Z_TAN_FV01_radial.nc')
+
+
+class TestAcornHandler(HandlerTestCase):
+    def setUp(self):
+        self.handler_class = AcornHandler
+        super(TestAcornHandler, self).setUp()
+
+    def test_good_netcdf_generic_fv00(self):
+        handler = self.run_handler(GOOD_NC_FV00,
+                                   include_regexes=['IMOS_ACORN_.*\.nc']
+                                   )
+        self.assertEqual(len(handler.file_collection), 1)
+        f = handler.file_collection[0]
+
+        self.assertEqual(f.dest_path, os.path.join('IMOS/ACORN/gridded_1h-avg-current-map_non-QC/ROT/2018/05/01',
+                                                   os.path.basename(GOOD_NC_FV00)))
+        self.assertEqual(f.publish_type, PipelineFilePublishType.HARVEST_UPLOAD)
+        self.assertTrue(f.is_checked)
+        self.assertTrue(f.is_stored)
+        
+    def test_fv00_in_dm(self):
+        # check that a FV00 file is not allowed in DM folder/pipeline
+        self.run_handler_with_exception(InvalidFileNameError, GOOD_NC_FV00,
+                                        include_regexes=[
+                                            '^IMOS_ACORN_[A-Z].*_[0-9]{8}T[0-9]{6}Z_[A-Z]{3,4}_FV01_(radial|sea-state|wavespec|windp|wavep|1-hour-avg)\.nc$']
```

Comment:
If I understand well, this regex will be set somewhere else in the pipeline. It would be good to have single entry point definition for this regex that could be re-used here.

Reply:
The regexp is set up here 
https://github.com/aodn/chef-private/pull/2842/files

Reply:
the one in the test_acorn is only there as a test, to force a config file in a way.

Reply:
Can you retrieve the regex from the .json rather than duplicating it in your unit test? Just in case we update the regex in the future...

Reply:
not really. The unittest have to be successful independently of the config.